packages/file-explorer/src/NewItemModal.tsx

```diff
@@ -294,9 +295,13 @@ class NewItemModal extends PureComponent<NewItemModalProps, NewItemModalState> {
 
   handleOverwriteConfirm(): void {
     this.setState({ showOverwriteModal: false });
-    const { onSubmit } = this.props;
+    const { onSubmit, onOverwrite } = this.props;
     const { path, value } = this.state;
     log.debug('handleOverwriteConfirm', path, value);
+
+    if (onOverwrite) {
+      onOverwrite(value);
+    }
     onSubmit(`${path}${value}`);
```

Comment:
Instead of adding an `onOverwrite` method, add a parameter `isOverwrite = false` (defaults to false) to `onSubmit`.